File: LAB2/2024-1/LAB2-2024_1-Desarrollo/SalaEspecializada.java
public class SalaEspecializada extends AmbienteClinico {
	private String nombre;
	private boolean poseeEquipamiento;
	
	public SalaEspecializada(double area, char torre, int piso, String nombre, boolean poseeEquipamiento) {
		super(area, torre, piso);
		this.nombre = nombre;
		this.poseeEquipamiento = poseeEquipamiento;
	}
	
	@Override
	public String devolverInformacion() {
		String cadena = "SALA ESPECIALIZADA:" + nombre + " - TORRE:" + torre + " - PISO:" + piso + " - EQ. IMAGENOLOGIA:";
		if(poseeEquipamiento == false)
			cadena += "NO\n";
		else
			cadena += "SI\n";
		return cadena;
	}
}

File: LAB4/2024-1/MedicalSoft/MedicalSoftModel/SalaEspecializada.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MedicalSoftModel
{
    public class SalaEspecializada : AmbienteClinico
    {
        private string nombre;
        private bool poseeEquipamientoImagenologia;
        private TipoSala tipoSala;
        public string Nombre { get => nombre; set => nombre = value; }
        public bool PoseeEquipamientoImagenologia { get => poseeEquipamientoImagenologia; set => poseeEquipamientoImagenologia = value; }
        public TipoSala TipoSala { get => tipoSala; set => tipoSala = value; }
        public override void mostrarDatos()
        {
            System.Console.WriteLine(IdAmbienteClinico + ". SALA ESPECIALIZADA:" + nombre + " - TORRE:" + Torre + " - PISO:" + Piso + " - EQ. IMAGENOLOGIA:" + (poseeEquipamientoImagenologia ? "SI" : "NO") + " - ESPACIO:" + EspacioMetrosCuadrados + " - TIPO SALA:" + TipoSala);
        }
    }
}
